modules/core/src/v2/coins/abstractUtxoCoin.ts

```diff
@@ -529,6 +530,12 @@ export abstract class AbstractUtxoCoin extends BaseCoin {
 
       // verify that each recipient from txParams has their own output
       const expectedOutputs = _.get(txParams, 'recipients', [] as TransactionRecipient[]);
+      if (wallet.baseCoin.getFamily() === 'bch') {
+        expectedOutputs.forEach((output) => {
+          const baseAddress = (wallet.baseCoin as Bch).canonicalAddress(output.address);
+          output.address = baseAddress;
+        });
+      }
```

Comment:
1. You can use `this.getFamily()` and `this.canonicalAddress()` instead
2. You should not mutate the object. Instead do this

```
const expectedOutputs = _.get(...).map((output) => {
  if (!isBch) {
    return output;
  }
  return { ...output, address: (this as Bch).canonicalAddress(output.address) };
});
```

Reply:
we should actually add this method to AbstractBaseCoin
```
canonicalAddress(address: string): string {
  return address;
}
```

then you can skip the `as BCH` casts and the conditional

Reply:
yes, Thanks for suggestion. I updated the all the changes as suggested.